### PR TITLE
add update resource

### PR DESCRIPTION
**Pull request by ningmingxiao**
Signed-off-by: ningmingxiao <ning.mingxiao@zte.com.cn>

### CODE REVIEW

File: cmd/nerdctl/update.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+func setUpdateFlags(cmd *cobra.Command) {
+	cmd.Flags().Float64("cpus", 0.0, "Number of CPUs")
+	cmd.Flags().Uint64("cpu-period", 0, "Limit CPU CFS (Completely Fair Scheduler) period")
+	cmd.Flags().Int64("cpu-quota", -1, "Limit CPU CFS (Completely Fair Scheduler) quota")
+	cmd.Flags().Uint64("cpu-shares", 0, "CPU shares (relative weight)")
+	cmd.Flags().StringP("memory", "m", "", "Memory limit")
+	cmd.Flags().String("cpuset-cpus", "", "CPUs in which to allow execution (0-3, 0,1)")
+	cmd.Flags().String("cpuset-mems", "", "MEMs in which to allow execution (0-3, 0,1)")
+}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/update.go:73]: Please update the Command Reference documentation like this

https://github.com/containerd/nerdctl/commit/19ca8c36a1f7e2e3bbc0e48a0d58954573f111f6

File: cmd/nerdctl/update.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+			return err
+		},
+	}
+	for _, req := range args {
+		n, err := walker.Walk(ctx, req)
+		if err != nil {
+			return err
+		} else if n == 0 {
+			return fmt.Errorf("no such container %s", req)
+		}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/update.go:98]: Also error out when `n > 1`

File: cmd/nerdctl/update.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+	if cStatus != "up" {
+		return nil
+	}
+	task, err := container.Task(ctx, nil)
+	if err != nil {
+		if errdefs.IsNotFound(err) {
+			// Task exited already.
+			return nil
+		}
+		return errors.Wrap(err, "failed to get task")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/update.go:243]: `fmt.Errorf` with `%w`

File: cmd/nerdctl/update.go (hunk 2 above)
Comment (ningmingxiao) [reply cmd/nerdctl/update.go:92]: update can update one or more containers

Comment (AkihiroSuda) [reply cmd/nerdctl/update.go:95]: Ambiguous ID should result in an error

File: README.md [hunk 4]
```diff
@@ -644,6 +645,20 @@ Usage: `nerdctl restart [OPTIONS] CONTAINER [CONTAINER...]`
 Flags:
 - :whale: `-t, --time=SECONDS`: Seconds to wait for stop before killing it (default "10")
 
+### :whale: nerdctl update
+Update configuration of one or more containers.
+
+Usage: `docker update [OPTIONS] CONTAINER [CONTAINER...]`
```
Comment (AkihiroSuda) [inline README.md:651]: Not docker

File: cmd/nerdctl/run_cgroup_linux_test.go [hunk 5]
```diff
@@ -66,6 +66,11 @@ func TestRunCgroupV2(t *testing.T) {
 		"cat", "cpu.max", "memory.max", "pids.max", "cpu.weight", "cpuset.cpus", "cpuset.mems").AssertOutExactly(expected)
 	base.Cmd("run", "--rm", "--cpu-quota", "42000", "--cpuset-mems", "0", "--cpu-period", "100000", "--memory", "42m", "--pids-limit", "42", "--cpu-shares", "2000", "--cpuset-cpus", "0-1", "-w", "/sys/fs/cgroup", testutil.AlpineImage,
 		"cat", "cpu.max", "memory.max", "pids.max", "cpu.weight", "cpuset.cpus", "cpuset.mems").AssertOutExactly(expected)
+
+	base.Cmd("run", "--name", "testUpdate", "-w", "/sys/fs/cgroup", "-d", testutil.AlpineImage, "top").AssertOK()
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run_cgroup_linux_test.go:70]:
```suggestion
	base.Cmd("run", "--name", testutil.Identifier(t)+"-testUpdate", "-w", "/sys/fs/cgroup", "-d", testutil.AlpineImage, "sleep", "infinity").AssertOK()
```

File: cmd/nerdctl/update.go (hunk 2 above)
Comment (ningmingxiao) [reply cmd/nerdctl/update.go:97]: ok.thank you

File: README.md [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
  ✅ Same UI/UX as `docker`
 
- ✅ Supports Docker Compose (`nerdctl compose up`)
+ ✅ Supports Docker Compose (`nerdctl compose up`)docker update [
```
Comment (AkihiroSuda) [inline README.md:12]: remove

File: cmd/nerdctl/update.go [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+	"github.com/containerd/containerd/pkg/cri/util"
+	"github.com/containerd/nerdctl/pkg/formatter"
+	"github.com/containerd/nerdctl/pkg/idutil/containerwalker"
+	"github.com/containerd/typeurl"
+	"github.com/docker/go-units"
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/spf13/cobra"
+)
+
+type updateRsourceOptions struct {
```
Comment (Junnplus) [inline cmd/nerdctl/update.go:38]: Resource ?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**